client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/backend/initialize/FeatureFlagsDto.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SonarLint Core - Client API
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.clientapi.backend.initialize;
+
+/**
+ * Optional features toggles of the backend. To ease transition or to accommodate different client needs.
+ */
+public class FeatureFlagsDto {
+
+  private final boolean shouldManageSmartNotifications;
+  @Deprecated
+  // not controllable anymore, it is the backend's responsibility to decide
+  private final boolean taintVulnerabilitiesEnabled;
+  private final boolean shouldSynchronizeProjects;
+  private final boolean shouldManageLocalServer;
+  private final boolean enableSecurityHotspots;
+
+  public FeatureFlagsDto(boolean shouldManageSmartNotifications, boolean taintVulnerabilitiesEnabled, boolean shouldSynchronizeProjects, boolean shouldManageLocalServer,
```

Comment:
It's a bit more work but wouldn't it be a good time to move to a capabilities system? Here the clients would still have to set each flag, and we could instead propose a list of opt-ins, with constants or enum values. See https://sonarsource.atlassian.net/browse/SLCORE-474

Reply:
You mean replacing booleans with enums? I think that even if adding new flags creates binary incompatibilities, this is not a big deal and clearly "document" new features that each IDE can decide to enable or not.
But we can do that later I guess.